loaders/httpfs/loader.go

```diff
@@ -0,0 +1,30 @@
+package httpfs
+
+import (
+	"io"
+	"net/http"
+)
+
+type httpFileSystemLoader struct {
+	http.FileSystem
+}
+
+func NewLoader(fs http.FileSystem) httpFileSystemLoader {
```

Comment:
I've seen the same arguments as mine in multiple times and also by some of the Go core members.
You can check here https://golang.org/doc/effective_go.html#generality.
Some of the problems with this is that golint will complain and i think that go vet also,  another problem is that go doc will not show the type because is unexported, and also make hard to people to use your type outside your package.